### PR TITLE
docs: Update changelog for v2.1.0 release

### PR DESCRIPTION
**Pull request by spencer-tb**
## 🗒️ Description
~~Updates the changelog for the bug fix release v2.0.1.~~
Updates the changelog for the bug fix release v2.1.0.

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](../docs/) directory.
- [x] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

File: docs/CHANGELOG.md
Comment (danceratopz) [inline docs/CHANGELOG.md:15]: If we want to be super-clean, then the new template should be added after we tag v2.0.1, otherwise it'll show-up at the top of the tagged doc version here:
https://ethereum.github.io/execution-spec-tests/v2.0.0/CHANGELOG/

**Review by danceratopz — COMMENTED** [review]
Minor nit, otherwise LGTM!

**Review by danceratopz — APPROVED** [review]
LGTM!

But, as mentioned, perhaps we can wait until #407 is merged and release v2.1.0 instead of v2.0.1 to avoid too many releases/updates in a short period of time?

File: docs/CHANGELOG.md
Comment (marioevz) [inline docs/CHANGELOG.md:7]:
```suggestion
## [v2.1.0](https://github.com/ethereum/execution-spec-tests/releases/tag/v2.1.0) - 2024-01-29: 🐍🏖️ Cancun
```

File: docs/CHANGELOG.md
Comment (marioevz) [inline docs/CHANGELOG.md:9]:
```suggestion
Release [v2.1.0](https://github.com/ethereum/execution-spec-tests/releases/tag/v2.1.0) fixes a small bug introduced within the previous release where transition forks are used within the new `StateTest` format. This was highlighted by @chfast within #405 (https://github.com/ethereum/execution-spec-tests/issues/405), where the fork name `ShanghaiToCancunAtTime15k` was found within the state tests.
```

**Review by marioevz — APPROVED** [review]
LGTM.

It seems there should be merge conflict between this and #407 because this deletes "### 🧪 Test Cases" from the changelog, and the other PR appends to it, but it's not showing up, so it might be worth it to try to rebase and see how it looks like.

**Pull request merged**
